### PR TITLE
Fix: search page order

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/6359

### Tests | QA Steps
1. Open the Search page.
2. Search for something.
3. Order of results should be https://github.com/Expensify/App/issues/6359#issuecomment-976021754.

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web | Android | Mobile Web | Desktop | iOS

https://user-images.githubusercontent.com/24370807/144112868-5bc20ed2-d861-4523-91a1-df487ce9dd99.mp4

### CODE REVIEW

**Review by Jag96 — COMMENTED** [review]
One thing I noticed is that the `#announce` rooms are still showing above some group chats:

| | |
|---|---|
| image | image |

For example in those the #announce room should be below the 1:1s and the group chats

**Comment by parasharrajat** [comment]
@Jag96 I have fixed it. Actually, we have to change the Initial proposal.
Now I am ordering the recent reports in the requested order based on the newly added parameter.
There is a known bug due to which room names are duplicated on each search. The good news is I have found the fix for that as well. I will propose it on the related issue.

**Comment by parasharrajat** [comment]
@Jag96 Also, could you please get  https://github.com/Expensify/App/issues/5699 rolling? No one is assigned to that issue.

File: src/libs/OptionsListUtils.js [hunk 1]
```diff
@@ -508,6 +509,18 @@ function getOptions(reports, personalDetails, activeReportID, {
         recentReportOptions = reportsSplitByDefaultChatRoom[0].concat(reportsSplitByDefaultChatRoom[1]);
     }
 
+    // If we are prioritizing 1:1 chats in search, do it only once we started searching
+    if (sortByReportTypeInSearch && searchValue !== '') {
+        recentReportOptions = lodashOrderBy(recentReportOptions, [(option) => {
+            if (option.isDefaultChatRoom || option.isArchivedRoom) {
+                return 3;
+            } if (!option.login) {
```
Comment (Jag96) [inline src/libs/OptionsListUtils.js:517]: Can you split this `if` to another line? I know it doesn't make a difference code-wise but it's a bit more consistent style-wise.

**Review by Jag96 — CHANGES_REQUESTED** [review]
Just one nitpick, otherwise, this looks good and works well!

File: src/libs/OptionsListUtils.js (hunk 1 above)
Comment (parasharrajat) [reply src/libs/OptionsListUtils.js:517]: Oops, it is supposed to be else if.

Comment (Jag96) [reply src/libs/OptionsListUtils.js:517]: No worries, eslint might say its a useless `else` b/c of return, if so feel free to put it on a new line

Comment (parasharrajat) [reply src/libs/OptionsListUtils.js]: just did.

**Review by Jag96 — APPROVED** [review]

**Pull request merged**